lib/discourse_activity_pub/user_handler.rb

```diff
@@ -140,5 +140,15 @@ def log_failure(verb, message)
       prefix = "Failed to #{verb} user for #{actor.id}"
       Rails.logger.warn("[Discourse Activity Pub] #{prefix}: #{message}")
     end
+
+    def update_avatar?
+      return false unless actor.icon_url
+      return true if !user || user.user_avatar.custom_upload.blank?
```

Comment:
Maybe a better check here is to use `user.uploaded_avatar`. That will return an upload object for either a custom upload or a gravatar, only letter avatar users will return `nil`.

Reply:
@pmusaraj Sure thing: https://github.com/discourse/discourse-activity-pub/pull/33/commits/fcb8c468d6b666d6599df6133ecee4465e84ce30